packages/adapter-cloudflare/src/worker.js

```diff
@@ -36,26 +37,30 @@ const worker = {
 				}
 			});
 		} else {
-			// prerendered pages and index.html files
-			pathname = pathname.replace(/\/$/, '') || '/';
-
-			let file = pathname.substring(1);
+			// prerendered pages and /static files
 
 			try {
-				file = decodeURIComponent(file);
-			} catch (err) {
-				// ignore
+				pathname = decodeURIComponent(pathname);
+			} catch {
+				// ignore invalid URI
 			}
 
-			if (
-				manifest.assets.has(file) ||
-				manifest.assets.has(file + '/index.html') ||
-				prerendered.has(pathname)
-			) {
+			const stripped_pathname = pathname.replace(/\/$/, '') || '';
```

Comment:
surely the only time you'd hit the `|| ''` case is if the returned value is _already_ `''`?

```suggestion
			const stripped_pathname = pathname.replace(/\/$/, '');
```